File: 16Feb2024.java
import java.util.Arrays;
import java.util.HashMap;

class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int a : arr)
            map.put(a, map.getOrDefault(a, 0) + 1);
        int[] fre = new int[map.size()];
        int i = 0;
        for (int key : map.keySet()) {
            fre[i] = map.get(key);
            i++;
        }
        Arrays.sort(fre);
        int ans = 0;
        for (i = 0; i < fre.length; i++) {
            if (k >= fre[i]) {
                k -= fre[i];
            } else {
                ans++;
            }
        }
        return ans;
    }
}

File: 16Feb2024.py
from ast import List


class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        mp = {}
        for a in arr :
            mp[a]=mp.get(a,0)+1
        list = []
        for fre in mp.values() :
            list.append(fre)
        ans=0
        list.sort()
        for i in range(len(list)) :
            if(k>=list[i]) :
                k-=list[i]
            else :
                ans+=1
        return ans